internal/core/data/application/event.go

```diff
@@ -58,20 +59,23 @@ func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (err error
 		}
 		e = addedEvent
 
-		lc.Debugf(
+		a.lc.Debugf(
 			"Event created on DB successfully. Event-id: %s, Correlation-id: %s ",
 			e.Id,
 			correlationId,
 		)
+
+		a.eventsPersistedCounter.Inc(1)
+		a.readingsPersistedCounter.Inc(int64(len(addedEvent.Readings)))
```

Comment:
sorry for the late review
the metrics should be collected after checking the config
```
if config.Writable.Telemetry.Metrics.EventsPersisted {
  a.eventsPersistedCounter.Inc(1)
}
if config.Writable.Telemetry.Metrics.ReadingsPersisted {
  a.readingsPersistedCounter.Inc(int64(len(addedEvent.Readings)))
}
```

Reply:
Nope, we are always collecting the metric. The configuration controls the reporting of the metrics to the MessageBus.

Reply:
@lenny-intel is there any advantage to collect them when the config is disabled?
I am not sure about the performance impact right now but we can test it.  Some of our customers use two or three 100 ms autoevent on a low power machine, so any additional CPU or memory consumption might cause issues.

Reply:
Yes, the advantage it simplicity of code. Doesn't take any logger to increment a counter than it takes to check the value of the Boolean setting